creator-node/src/utils/tracing.ts

```diff
@@ -0,0 +1,86 @@
+import type { Span, SpanOptions, SpanContext } from '@opentelemetry/api'
+import { trace, context, SpanStatusCode } from '@opentelemetry/api'
+import { SemanticAttributes } from '@opentelemetry/semantic-conventions'
+import { getTracer } from '../tracer'
+
+/**
+ * Higher-order function that adds opentelemtry tracing to a function
+ */
+export const instrumentTracing = <TFunction extends (...args: any[]) => any>({
+  name,
+  fn,
+  options
+}: {
+  name?: string
+  fn: TFunction
+  options?: SpanOptions
+}) => {
+  return (...args: Parameters<TFunction>): ReturnType<TFunction> => {
+    const spanName = name || fn.name
+    const spanOptions = options || {}
+    return getTracer().startActiveSpan(spanName, spanOptions, (span: Span) => {
+      try {
+        span.setAttribute(SemanticAttributes.CODE_FUNCTION, fn.name)
+        // TODO: add skip parameter to instrument testing function to NOT logo certain args
+        // span.setAttribute('args', JSON.stringify(args))
+        return fn(...args)
+      } catch (e) {
+        span.recordException(e as Error)
+        span.setStatus({ code: SpanStatusCode.ERROR })
+        throw e
+      } finally {
+        span.end()
+      }
+    })
+  }
+}
+
+/**
+ * Helper function that gets the current active span or return `undefined` if there is no active span
+ */
+export const getActiveSpan = (): Span | undefined => {
+  return trace.getSpan(context.active())
+}
+
+export const info = (msg: string) => {
```

Comment:
could we namespace all of these with tracing? like `recordException` or `warn` by themselves might be a bit confusing standalone in downstream code

Reply:
Do you have a recommendation for a name? Ideally it'd be something like `logger` or `tracing` but there might be name conflicts there